### ISSUE TITLE
[dev.icinga.com #11892] Livestatus custom_variables filter does not work

### ISSUE THREAD

**Issue by icinga-migration** [Icinga/icinga2#4256]
This issue has been migrated from Redmine: https://dev.icinga.com/issues/11892

**Created by sni on 2016-06-03 10:19:04 +00:00**

Assignee: _(none)_
Status: _New_ 
Target Version: _(none)_
Last Update: _2016-06-06 08:24:00 +00:00 (in Redmine)_

    Icinga Version: r2.4.10-1
    Backport?: Not yet backported
    Include in Changelog: 1

---

    $ cat query.txt
    GET hosts
    Filter: custom_variables ~~ TAGS test
    Stats: name !=
    OutputFormat: json
    ResponseHeader: fixed16

    $ cat query.txt | nc -U .../live
    452        3144
    Error: Invalid operator for column 'custom_variables': ~~ (expected '>=' or '=').

            (0) liblivestatus.so: void boost::throw_exception >(boost::exception_detail::error_info_injector const&) (+0xf4) [0x7f3f15bb2c74]
            (1) liblivestatus.so: void boost::exception_detail::throw_exception_(std::invalid_argument const&, char const*, char const*, int) (+0x69) [0x7f3f15bb2d29]
            (2) liblivestatus.so: icinga::AttributeFilter::Apply(boost::intrusive_ptr const&, icinga::Value const&) (+0xe9f) [0x7f3f15b9e7df]
            (3) liblivestatus.so: icinga::AndFilter::Apply(boost::intrusive_ptr const&, icinga::Value const&) (+0x2f) [0x7f3f15b4cd1f]
            (4) liblivestatus.so: icinga::Table::FilteredAddRow(std::vector >&, boost::intrusive_ptr const&, int, icinga::Value const&, icinga::LivestatusGroupByType, boost::intrusive_ptr const&) (+0x73) [0x7f3f15b85be3]
            (5) liblivestatus.so: boost::detail::function::function_obj_invoker3 >&, boost::intrusive_ptr const&, int, icinga::Value const&, icinga::LivestatusGroupByType, boost::intrusive_ptr const&>, boost::_bi::list7, boost::reference_wrapper > >, boost::_bi::value >, boost::_bi::value, boost::arg<1>, boost::arg<2>, boost::arg<3> > >, bool, icinga::Value const&, icinga::LivestatusGroupByType, boost::intrusive_ptr const&>::invoke(boost::detail::function::function_buffer&, icinga::Value const&, icinga::LivestatusGroupByType, boost::intrusive_ptr const&) (+0x3d) [0x7f3f15ba512d]
            (6) liblivestatus.so:  (+0x5372e) [0x7f3f15b5372e]
            (7) liblivestatus.so: icinga::HostsTable::FetchRows(boost::function const&)> const&) (+0x1b3) [0x7f3f15b8a023]
            (8) liblivestatus.so: icinga::Table::FilterRows(boost::intrusive_ptr const&, int) (+0x176) [0x7f3f15b51426]
            (9) liblivestatus.so: icinga::LivestatusQuery::ExecuteGetHelper(boost::intrusive_ptr const&) (+0x104) [0x7f3f15ba1ae4]
            (10) liblivestatus.so: icinga::LivestatusQuery::Execute(boost::intrusive_ptr const&) (+0x1f1) [0x7f3f15ba2ea1]
            (11) liblivestatus.so: icinga::LivestatusListener::ClientHandler(boost::intrusive_ptr const&) (+0x1b0) [0x7f3f15ba32a0]
            (12) libbase.so: icinga::ThreadPool::WorkerThread::ThreadProc(icinga::ThreadPool::Queue&) (+0x359) [0x7f3f1be3b9b9]
            (13) libboost_thread.so.1.55.0:  (+0xdaea) [0x7f3f1cabfaea]
            (14) libpthread.so.0:  (+0x80a4) [0x7f3f1c10d0a4]
            (15) libc.so.6: clone (+0x6d) [0x7f3f195eb87d]

The only reference i found is:
http://lists.mathias-kettner.de/pipermail/checkmk-commits/2011-June/001206.html

Basically the syntax is:
Filter: custom\_variables <OP> <CUST_VAR_NAME> <CUST_VAR_VALUE>

See also: https://github.com/sni/Thruk/issues/552

**Comment by dnsmichi** [comment]
Unfortunately we won't put any effort into this feature anymore.
